### ISSUE TITLE
can't install throught `pip install -q git+https://github.com/tensorflow/docs`

### ISSUE THREAD

**Issue by danyow-cheung** [tensorflow/tensorflow#59747]
seen as below 
![error](https://user-images.githubusercontent.com/76671016/220236110-8b97769f-f846-447a-ae6e-bddf90a6bb05.JPG)
i got this command from here:<u>https://keras.io/examples/vision/video_classification/</u>

tensorflow version : 2.11.0

**Comment by Sarvesh-Kesharwani** [comment]
![Code_cpIHjnPyDM](https://user-images.githubusercontent.com/22115102/220269593-fd7c9f83-7730-42c9-8b42-8f0762ef31bd.png)

It is working for me in windows 11 with the above shown versions, please specify more details about your OS, etc.
so that we can reproduce this error.

**Comment by danyow-cheung** [comment]
desktop;Windows 10 
pip version:23.0.1 
python :3.7 


here is all the libray in the virtual env 
------------------------------------

absl-py==1.3.0
astunparse==1.6.3
attrs==22.1.0
backcall==0.2.0
cachetools==5.2.1
certifi==2022.12.7
charset-normalizer==2.1.1
colorama==0.4.6
cycler==0.11.0
debugpy==1.6.6
decorator==5.1.1
entrypoints==0.4
flatbuffers==22.11.23
fonttools==4.38.0
gast==0.4.0
google-auth==2.16.0
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
grpcio==1.51.1
h5py==3.7.0
idna==3.4
imageio==2.25.1
importlib-metadata==6.0.0
imutils==0.5.4
ipykernel==6.16.2
ipython==7.34.0
jedi==0.18.2
joblib==1.2.0
jupyter_client==7.4.9
jupyter_core==4.12.0
kaggle==1.5.12
keras==2.11.0
kiwisolver==1.4.4
libclang==15.0.6.1
Markdown==3.4.1
MarkupSafe==2.1.1
matplotlib==3.5.3
matplotlib-inline==0.1.6
mediapipe==0.9.0
nest-asyncio==1.5.6
numpy==1.21.6
oauthlib==3.2.2
opencv-contrib-python==4.6.0.66
opt-einsum==3.3.0
packaging==21.3
pandas==1.3.5
parso==0.8.3
pickleshare==0.7.5
Pillow==9.3.0
prompt-toolkit==3.0.36
protobuf==3.19.6
psutil==5.9.4
pyasn1==0.4.8
pyasn1-modules==0.2.8
Pygments==2.14.0
pyparsing==3.0.9
python-dateutil==2.8.2
python-slugify==8.0.0
pytz==2022.7.1
pywin32==305
pyzmq==25.0.0
requests==2.28.1
requests-oauthlib==1.3.1
rsa==4.9
scikit-learn==1.0.2
scipy==1.7.3
six==1.16.0
tensorboard==2.11.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorflow==2.11.0
tensorflow-estimator==2.11.0
tensorflow-intel==2.11.0
tensorflow-io-gcs-filesystem==0.29.0
termcolor==2.2.0
text-unidecode==1.3
threadpoolctl==3.1.0
tornado==6.2
tqdm==4.64.1
traitlets==5.9.0
typing_extensions==4.4.0
urllib3==1.26.14
wcwidth==0.2.6
Werkzeug==2.2.2
wrapt==1.14.1
zipp==3.11.0

> ![Code_cpIHjnPyDM](https://user-images.githubusercontent.com/22115102/220269593-fd7c9f83-7730-42c9-8b42-8f0762ef31bd.png)
> 
> It is working for me in windows 11 with the above shown versions, please specify more details about your OS, etc. so that we can reproduce this error.

**Comment by tilakrayal** [comment]
@danyow-cheung,
I tried to execute the code **!pip install -q git+https://github.com/tensorflow/docs** from the mentioned official keras doc  on both Windows and colab and it was executed without any issue/error. Kindly find the gist of it [here](https://colab.sandbox.google.com/gist/tilakrayal/8fd4862f3b50b725ddb94e4b687ac2ab/untitled977.ipynb) and the SS for the reference.

![image](https://user-images.githubusercontent.com/81610181/220498184-f48a89b3-f911-4561-8bfe-31158c855a01.png)
![image](https://user-images.githubusercontent.com/81610181/220498294-9fae67a8-c39d-4e43-8cfe-ca66d4d3da08.png)
  
Could you please create a virtual environment and test your code again. Thank you!

**Comment by danyow-cheung** [comment]
> @danyow-cheung, I tried to execute the code **!pip install -q git+https://github.com/tensorflow/docs** from the mentioned official keras doc on both Windows and colab and it was executed without any issue/error. Kindly find the gist of it [here](https://colab.sandbox.google.com/gist/tilakrayal/8fd4862f3b50b725ddb94e4b687ac2ab/untitled977.ipynb) and the SS for the reference.
> 
> ![image](https://user-images.githubusercontent.com/81610181/220498184-f48a89b3-f911-4561-8bfe-31158c855a01.png) ![image](https://user-images.githubusercontent.com/81610181/220498294-9fae67a8-c39d-4e43-8cfe-ca66d4d3da08.png)
> 
> Could you please create a virtual environment and test your code again. Thank you!

still the same,but i will try my code on colab next time,thank you 
![error](https://user-images.githubusercontent.com/76671016/220501511-85d8f0c9-6c3e-4b22-99d3-e3aef0960d7a.JPG)

**Comment by tilakrayal** [comment]
@danyow-cheung,
Yeah, please try to execute in colab and update the same. And the error clearly mentioned that it is not the problem with either pip or tensorflow. Also this particular error is caused by not having a C/C++ compiler installed. As said in the error message, either install [MSVC](https://visualstudio.microsoft.com/visual-cpp-build-tools/) or another compiler to compile it.

Please try to download Visual Studio from [its main page](https://visualstudio.microsoft.com/).

If Visual Studio is already installed, then when you run the installer, you can modify it (by clicking the modify button). Thank you!

**Comment by tilakrayal** [comment]
Closing this as stale. Please reopen if this is still a valid request. Thank you!
